app.go

```diff
@@ -368,6 +384,7 @@ type Application struct {
 	config                      *cctypes.Config
 	childTxRefs                 []evmaux.ChildTxRef // links Tendermint txs to EVM txs
 	ReceiptsVersion             int32
+	EVMState                    *gstate.StateDB
```

Comment:
Does this `StateDB` instance really need to be in `Application`, why can't the `EvmStore` own it?

Reply:
The name is misleading. StateDB is actually a state not DB. The wrapper is like StateDB -> TrieDB -> EvmStore.

Reply:
Yeah, so maybe we should have something like this?
```golang
struct EVMState {
  sdb StateDB
  evmStore *EvmStore

  Commit() error {
    evmStateRoot, err := sdb.Commit(true)
    if err != nil {
      return nil, err
    }
    evmStore.SetVMRootKey(evmStateRoot[:])
    sdb.Reset(evmStateRoot)
  }

  GetSnapshot(version int64) (*EVMState, error) {
    evmStateRoot, err := evmStore.GetRootAt(version)
    if err != nil {
      return err
    }
    stateDB, err := gstate.New(gcommon.BytesToHash(evmStateRoot), sdb.Database())
    if err != nil {
      return nil, err
    }
    return &EVMState{ sdb: stateDB, evmStore: evmStore }
  }
}
```